src/components/ScrollTo/ScrollTo.stories.tsx

```diff
@@ -7,54 +7,21 @@ export default {
     component: Component,
 };
 
-function Spam(n: number, dir: string) {
-    // react use state
-    // generate an array of spam
-    // remove doc*
-    // click on button => add to list, similar to sectionlist
-    // see https://github.com/c2c-project/c2c-chat/blob/master/client/src/components/chat/chat.stories.jsx
-    // for inspiration
-
-    const [spam, updateSpam] = React.useState(n);
-    React.useEffect(() => {
-        const effect = document.getElementById('scrollTo');
-        const toAdd =
-            "<div style={{ height: '500px' }}><h1>SPAM #`${spam}`</h1></div>";
-        if (dir === 'top') {
-            effect?.insertAdjacentHTML('beforebegin', toAdd);
-        }
-    });
-
-    // const doc = document.getElementById('scrollTo');
-    // let toAdd = '<p><br />';
-    // const component = '<Component active >a</Component>';
-    // // needed as we cannot use n in the loop param if we modify it
-    // let x = n;
-    // while (x > 0) {
-    //     // eslint-disable-next-line
-    //     toAdd = toAdd + 'spam' + '<br />';
-    //     x -= 1;
-    // }
-    // toAdd += '</p>';
-    // if (dir === 'top') {
-    //     doc?.insertAdjacentHTML('beforebegin', toAdd);
-    //     doc?.insertAdjacentHTML('beforebegin', component);
-    // } else if (dir === 'bottom') {
-    //     doc?.insertAdjacentHTML('beforeend', toAdd);
-    //     doc?.insertAdjacentHTML('beforeend', component);
-    // } else {
-    //     doc?.insertAdjacentHTML('afterbegin', toAdd);
-    // }
-    // const temp = document.createElement('p');
-    // temp.insertAdjacentHTML('beforeend', toAdd);
-    // document.body.appendChild(temp);
-}
-
 export function ScrollToTop() {
+    const [cnt, setCount] = React.useState(0);
+    const [spam, updateSpam] = React.useState<JSX.Element[]>([<span key={cnt}><br/><br/><br/><h1>SPAM</h1><br/><br/><br/></span>]);
+    const update = () => {
+        setCount(cnt + 1);
+        spam.push(<span key={cnt}><br/><h1>SPAM</h1><br/></span>);
+        updateSpam(spam);
+    }
+    
     return (
         <div id='scrollTo'>
             <Component active direction='top'>
-                <button type='button' onClick={() => Spam(1000, 'top')}>
+                <p>{spam}</p>
+                <p>{cnt}</p>
+                <button type='button' onClick={() => update()}>
                     click to add spam
                 </button>
             </Component>
```

Comment:
Although unnecessary, an optimization could be made here. I'm going to write it out for learning purposes, but there's no need to make the change
```js
function Example() {
    const [spam, setSpam] = React.useState<string[]>([]);

    const update = (previousSpam: string[]) => {
        const newSpam = [...previousSpam];
        for (let i = 0; i < 100; i += 1) {
            newSpam.push('');
        }
        return newSpam;
    };

    return (
        <div>
            <button type='button' onClick={() => setSpam(update)}>
                Add Spam
            </button>
            {spam.map(() => (
                <div>spam</div>
            ))}
        </div>
    );
}

```
Notice we only need to use one state and pass in a function to the `setSpam` function.  Remember from React documentation, we can pass in a function to the `setState` function whose sole argument will be the previous state, which we can then use to make the next state. We could also use the `update` function for both of our stories by moving it outside of the `Example` function component.  

Then, we achieve the following:
```js

const update = (previousSpam: string[]) => {
    const newSpam = [...previousSpam];
    for (let i = 0; i < 100; i += 1) {
        newSpam.push('');
    }
    return newSpam;
};

function ScrollToTop() {
    const [spam, setSpam] = React.useState<string[]>([]);

    return (
        <div>
            <button type='button' onClick={() => setSpam(update)}>
                Add Spam
            </button>
            {spam.map(() => (
                <div>spam</div>
            ))}
        </div>
    );
}

function ScrollToBottom() {
    const [spam, setSpam] = React.useState<string[]>([]);

    return (
        <div>
            <button type='button' onClick={() => setSpam(update)}>
                Add Spam
            </button>
            {spam.map(() => (
                <div>spam</div>
            ))}
        </div>
    );
}

```

Reply:
Awesome! It works now, BUT I keep getting this error on the `spam.map` section. 
```
Type 'Element[]' is not assignable to type 'Element'.ts(2322)
```
The story works as intended but the error is still there.